waspc/data/Generator/templates/server/src/auth/utils.ts

```diff
@@ -12,3 +18,72 @@ export const authConfig = {
   failureRedirectPath: "{= failureRedirectPath =}",
   successRedirectPath: "{= successRedirectPath =}",
 }
+
+export async function findUserBy<K extends keyof {= userEntityUpper =}>(where: { [key in K]: {= userEntityUpper =}[K] }): Promise<{= userEntityUpper =}> {
+  return prisma.{= userEntityLower =}.findUnique({ where });
+}
+
+export async function createUser(data: {= userEntityUpper =}): Promise<{= userEntityUpper =}> {
+  try {
+    return await prisma.{= userEntityLower =}.create({ data })
+  } catch (e) {
+    if (e instanceof AuthError) {
+      throw new HttpError(422, 'Validation failed', { message: e.message })
+    } else if (isPrismaError(e)) {
+      throw prismaErrorToHttpError(e)
+    } else {
+      throw new HttpError(500)
+    }
+  }
+}
+
+export async function createAuthToken(user: {= userEntityUpper =}): Promise<string> {
+  return sign(user.id);
+}
+
+export async function verifyToken(token: string): Promise<{ id: any }> {
+  return verify(token);
+}
+{=# isEmailAuthEnabled =}
+export async function updateUserEmailVerification(userId: {= userEntityUpper =}Id): Promise<void> {
+  try {
+    await prisma.{= userEntityLower =}.update({
+      where: { id: userId },
+      data: { isEmailVerified: true },
+    })
+  } catch (e) {
+    if (e instanceof AuthError) {
+      throw new HttpError(422, 'Validation failed', { message: e.message })
+    } else if (isPrismaError(e)) {
+      throw prismaErrorToHttpError(e)
+    } else {
+      throw new HttpError(500)
+    }
+  }
+}
+
+export async function updateUserPassword(userId: {= userEntityUpper =}Id, password: string): Promise<void> {
```

Comment:
Just seeing this update function made me wonder if we provide any methods to update other user fields. We may want to put some restrictions in auth validations on updating the `User.email` since then they could take over any other email after they verified theirs.

Reply:
Changing the email should imply that the user must verify the e-mail again. That's true!

Do we want to expose a function "updateUserEmail" for the user to use or somehow do it inside of Prisma middleware? 🤷

Reply:
Additional option: provide an option `allowEmailModification` which is by default `false`, and based on that flag we implement Prisma middleware.